firebase-firestore/src/main/java/com/google/firebase/firestore/model/SnapshotVersion.java

```diff
@@ -36,6 +37,12 @@ public Timestamp getTimestamp() {
     return timestamp;
   }
 
+  /** Returns the microseconds since EPOCH that this snapshot version represents. */
+  public long toMicroseconds() {
```

Comment:
Historically we've tried to avoid baking in this assumption about backend timestamp resolution into our system. This is because the backend *currently* truncates this time but they have an open bug for fixing this.

Meanwhile we've already established a precedent for how to represent timestamps literally in our schema. For example, the targets table includes the following:

```
                  + "snapshot_version_seconds INTEGER, "
                  + "snapshot_version_nanos INTEGER, "
```

It seems like we lose nothing by representing these accurately--even if we only query on seconds we'll be fine for the purposes of index-free querying--so it seems like we should follow the established precedent and represent timestamps at full resolution.

Reply:
I updated this PR to use the same seconds/nanos approach as the targets table. I didn't want to do this originally cause it would result in some overselection, but I assume that you are correct when you say that we don't need to worry about this here.